src/client-side-encryption/auto_encrypter.ts

```diff
@@ -473,8 +474,19 @@ export class AutoEncrypter {
   /**
    * Decrypt a command response
    */
-  async decrypt(response: Uint8Array | Document, options: CommandOptions = {}): Promise<Document> {
-    const buffer = Buffer.isBuffer(response) ? response : serialize(response, options);
+  async decrypt(
+    response: Uint8Array | Document | MongoDBResponse,
+    options: CommandOptions = {}
+  ): Promise<Document> {
+    const buffer = MongoDBResponse.is(response)
+      ? response.toBytes()
+      : Buffer.isBuffer(response)
```

Comment:
I don't feel strongly, but BSON's types specify `Uint8Array`s but detect runtime and return `Buffer`s if running in Nodejs.  Since our FLE bindings only support Nodejs, using `Buffer` APIs is safe

Reply:
"Safe" within some degree of assumption about the environment you are running in. I can make BSON in Node.js return Uint8Arrays:

```js
Object.defineProperty(Buffer.prototype, '_isBuffer', {value: true}); 
Buffer.isBuffer(require('bson').serialize({a:1}));
```